src/services/azureKeyVaultService.test.ts

```diff
@@ -0,0 +1,112 @@
+import Serverless from "serverless";
+import { MockFactory } from "../test/mockFactory";
+import {
+  AzureKeyVaultService,
+  AzureKeyVaultConfig
+} from "./azureKeyVaultService";
+
+import { KeyVaultManagementClient, Vaults } from "@azure/arm-keyvault";
+import { FunctionAppService } from "./functionAppService";
+
+describe("Azure Key Vault Service", () => {
+  const options: Serverless.Options = MockFactory.createTestServerlessOptions();
+  const createOrUpdate = jest.fn(() => {
+    console.log();
+  });
+  const knownVaults = {
+    testVault: {
+      location: "WestUS",
+      properties: {
+        accessPolicies: []
+      }
+    }
+  };
+
+  let serverless: Serverless = MockFactory.createTestServerless();
+
+  beforeEach(() => {
+    // KeyVaultManagementClient.prototype.vaults = {
+    //   get: jest.fn(async (resourceGroup, vaultName) => {
+    //     if (!knownVaults.hasOwnProperty(vaultName)) {
+    //       throw new Error("No matching vaults");
+    //     }
+    //     return knownVaults["testVault"];
+    //   }),
+    //   createOrUpdate,
+    // } as any;
+
+    Vaults.prototype.createOrUpdate = jest.fn();
+    Vaults.prototype.get = jest.fn(async (resourceGroup, vaultName) => {
+      if (!knownVaults.hasOwnProperty(vaultName)) {
+        throw new Error("No matching vaults");
+      }
+      return knownVaults["testVault"];
+    }) as any;
+
+    FunctionAppService.prototype.get = jest.fn(() => {
+      console.log("testing");
+      return {
+        identity: {
+          tenantId: "tid",
+          principalId: "oid"
+        }
+      } as any;
+    });
+  });
+
+  it("is defined", () => {
+    expect(AzureKeyVaultService).toBeDefined();
+  });
+
+  it("can be instantiated", () => {
+    const service = new AzureKeyVaultService(serverless, options);
+    expect(service).not.toBeNull();
+  });
+
+  it("Throws an error if correct keyvault name is not specified", async () => {
```

Comment:
Test name seems off.  The test shows that the vault doesn't exist - not that it was not specified.